File: src/main/java/edu/eci/arsw/blacklistvalidator/PerformanceMain.java
package edu.eci.arsw.blacklistvalidator;

public class PerformanceMain {

    public static void main(String[] args) {

        HostBlackListsValidator validator = new HostBlackListsValidator();
        String ip = "202.24.34.55";

        int cores = Runtime.getRuntime().availableProcessors();

        int[] threadConfigurations = {
                1,
                cores,
                cores * 2,
                50,
                100,
                1000
        };

        System.out.println("Available cores: " + cores);
        System.out.println("--------------------------------");

        for (int threads : threadConfigurations) {

            long startTime = System.currentTimeMillis();

            validator.checkHost(ip, threads);

            long endTime = System.currentTimeMillis();

            System.out.println(
                    "Threads: " + threads +
                            " | Execution time: " + (endTime - startTime) + " ms"
            );
            //Pausa para diferenciar rendimientos en VisualVM
            try {
                Thread.sleep(3000);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        }

    }
}

File: src/main/go/performance_main.go
package main

import (
	"blacklist-search-go/blacklistvalidator"
	"fmt"
	"runtime"
	"time"
)

func main() {
	validator := blacklistvalidator.HostBlackListsValidator{}
	ip := "202.24.34.55"

	cores := runtime.NumCPU()

	threadConfigurations := []int{
		1,
		cores,
		cores * 2,
		50,
		100,
		1000,
		10000,
		100000,
	}

	fmt.Println("Available cores:", cores)
	fmt.Println("--------------------------------")

	for _, threads := range threadConfigurations {
		start := time.Now()

		validator.CheckHost(ip, threads)

		elapsed := time.Since(start)

		fmt.Printf(
			"Goroutines: %d | Execution time: %d ms\n",
			threads,
			elapsed.Milliseconds(),
		)

		// Pausa para observar métricas
		time.Sleep(3 * time.Second)
	}
}
